src/engine/QueryPlanner.h

```diff
@@ -246,7 +246,6 @@ class QueryPlanner {
       const std::string& right);
 
   std::string generateUniqueVarName();
-  std::string toUniqueVariable(const string& entity);
```

Comment:
I am really sorry, this was neither finished nor working, and never intended to be pushed to the master branch. I owe you one for this.

Reply:
No problem. I quite like the idea, but right now all path operators assume inputs of size 2, which would need to be changed for this to work.